### PR TITLE
Support building with gcc-4.8.

### PR DESCRIPTION
**Pull request by coryan**
Create configuration files to build with gcc-4.8, the oldest compiler we are planning to support. A separate internal CL will enable this build.

### CODE REVIEW

**Comment by coryan** [comment]
@devjgm I need some help to merge this change.  The code fails with gcc-4.8, but I cannot figure out why. Can you take a look?  If not, that is cool.

```
==================== Test output for //google/cloud/spanner:spanner_client_value_test:
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from Value
[ RUN      ] Value.BasicSemantics
google/cloud/spanner/value_test.cc:43: Failure
Expected equality of these values:
  init
    Which is: { optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false} }
  *v.get<T>()
    Which is: { optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{false}, optional<bool>{false} }
Google Test trace:
google/cloud/spanner/value_test.cc:76: Testing: vector<optional<bool>> 0
google/cloud/spanner/value_test.cc:72: Testing: vector<bool> 0
google/cloud/spanner/value_test.cc:70: Testing: bool 0
google/cloud/spanner/value_test.cc:44: Failure
Expected equality of these values:
  init
    Which is: { optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false}, optional<bool>{false} }
  static_cast<T>(v)
    Which is: { optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{true}, optional<bool>{false}, optional<bool>{false} }
Google Test trace:
google/cloud/spanner/value_test.cc:76: Testing: vector<optional<bool>> 0
google/cloud/spanner/value_test.cc:72: Testing: vector<bool> 0
google/cloud/spanner/value_test.cc:70: Testing: bool 0
[  FAILED  ] Value.BasicSemantics (5 ms)
[ RUN      ] Value.DoubleNaN
[       OK ] Value.DoubleNaN (0 ms)
[ RUN      ] Value.MixingTypes
[       OK ] Value.MixingTypes (0 ms)
[ RUN      ] Value.SpannerArray
[       OK ] Value.SpannerArray (1 ms)
[----------] 4 tests from Value (6 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (6 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Value.BasicSemantics

 1 FAILED TEST
================================================================================
```

**Comment by devjgm** [comment]
FTR, fixed in https://github.com/googleapis/google-cloud-cpp/pull/2757

**Comment by coryan** [comment]
PTAL. The build passes locally, will enable it once this is merged.

File: google/cloud/spanner/value_test.cc [hunk 1]
```diff
@@ -22,6 +22,15 @@
 
 namespace google {
 namespace cloud {
+
+void PrintTo(optional<bool> const& v, std::ostream* os) {
```
Comment (devjgm) [inline google/cloud/spanner/value_test.cc]: Do you want to keep the edits in this file, or revert them?

Comment (coryan) [reply google/cloud/spanner/value_test.cc]: Reverting them seems like a better idea. Done.

**Comment by coryan** [comment]
PTAL.

**Pull request merged**